app/models/release_platform_run.rb

```diff
@@ -201,7 +201,8 @@ def last_commit
   end
 
   def finished_steps?
-    last_commit&.step_runs&.where(release_platform_run: self)&.success&.size == steps.size
+    return false if release_platform.release_step.blank?
+    last_commit&.step_runs&.where(release_platform_run: self, step: release_platform.release_step)&.first&.success?
```

Comment:
I think we can early check the lack of `last_commit` to avoid too many of these safe navigators.